Wrap up design pass on theming

vdonato opened the pull request:
NOTE: The files changed number looks pretty horrifying, but they're essentially all just snapshot changes caused by some slight color changes with broad impact. The code changes mostly involve tweaking colors and aren't very involved.

This PR builds off #2906 and does the last bit of work necessary to get it merged, including 
* updating snapshots and fixing other tests
* lightly refactoring the code that computes derived colors
* finishing up removing the `secondaryColor` theme option


Here's #2906's original PR description written by @tvst :
> I did a design pass on our themes, and in the process of cleaning up our colors Abhi and I realized that it would be better if we replaced `secondaryColor` with a set of calculated colors. So what this PR does is:
> 
> 1. **Tweaks a bunch of colors.** See [the checked-off items on this page](https://www.notion.so/streamlit/Approve-Product-Launch-a003dcf5cee14d24b5aacda8ae60aeaf#27bbf22fb22449ad9be2703026550d16) for a subset of the changes.
> 2. **Removes `secondaryColor`**.
> 3. **Cleans up the settings dialog a little bit.** It didn't do much to the theme editor, though. I believe Abhi is going to sync with Vincent on additional improvements there.
> 
> You can see the end results [here](https://www.notion.so/streamlit/Approve-Product-Launch-a003dcf5cee14d24b5aacda8ae60aeaf#d045e609425f43ec92e8762554968728)
> 
> NOTE:
> 
> 1. This PR has no tests!!! It's a "design PR" laughing
> 2. See `/src/theme/utils.ts` line 292. I left a question in there about what variable names we should use.

tvst committed:
style scrollbars

tvst committed:
Fix header gradient color

tvst committed:
Style menu and dialogs

tvst committed:
Style everything else!

tvst committed:
Fix small text in dialog

tvst committed:
Make button border use fadedText10, like other borders.

tvst committed:
Fix slider active color.

tvst committed:
Improve "made with streamlit" text

vdonato committed:
Remove remaining usage of secondaryColor

vdonato committed:
Update enzyme snapshots for SettingsDialog and ThemeCreator

vdonato committed:
Get rid of StyledThemeColorPicker

We can just use the ColorPicker directly now that we don't have any styles to
override for it.

vdonato committed:
Fix (non-snapshot) js unit tests

vdonato committed:
Don't use primary color for background when no radio options

vdonato committed:
Don't use primary color for border when checkbox disabled

vdonato committed:
Update a TON of snapshots

vdonato committed:
Use fadedText colors more where equivalent

vdonato committed:
Move derived color computation into a helper and other misc cleanup

vdonato committed:
Use shorthand style props for borders where possible

kantuni commented on frontend/src/components/elements/VegaLiteChart/VegaLiteChart.tsx:
```diff
@@ -493,22 +495,9 @@ function configWithThemeDefaults(config: any, theme: Theme): any {
-      ...config.title,
-    },
-  }
+  const mergedTheme = {}
```
Nit: IMHO code that doesn't modify things looks a lot cleaner. 
```js
return merge({}, themeDefaults, config || {})
```

vdonato replied:
fixed :+1:

kantuni commented on frontend/src/components/shared/Radio/Radio.tsx:
```diff
@@ -83,13 +83,21 @@ class Radio extends React.PureComponent<Props, State> {
                   }),
                 },
+                RadioMarkOuter: {
+                  style: ({ $checked }: { $checked: boolean }) => ({
```
Nit: There should exist a type for Radio with `{ $checked: boolean }` in Base. Try using that instead.

vdonato replied:
Can't do this for Radio as it seems like the [override types for that component are lazy](https://github.com/uber/baseweb/blob/2872bd88fea0afdf3dabb93f10e7e6ef620957b2/src/radio/index.d.ts#L79) and don't actually export anything for this :disappointed: 

Will do for `Accordion` in `withExpander`, though

kantuni commented on frontend/src/hocs/withExpandable/withExpandable.tsx:
```diff
@@ -118,7 +118,7 @@ function withExpandable(
             ToggleIcon: {
               style: ({ $disabled }) => ({
```
same here

vdonato replied:
done!

kantuni commented on frontend/src/theme/utils.ts:
```diff
@@ -310,17 +337,14 @@ export const createEmotionTheme = (
-    },
-    {}
-  )
+  const parsedColors: Record<string, string> = {}
```
I would vote to keep the old `reduce` here.

vdonato replied:
Will do. This one's a bit weird since I generally prefer using `reduce`, but semantically you'd expect to see the reducer function return a new accumulator object instead of mutating the one given to it, and doing that here seemed like overkill given how much more verbose it ends up being.

But yeah, it's a minor enough thing that I'm happy leaving this unchanged for now / can file another PR to have this follow more normal reduce semantics

kantuni commented on lib/streamlit/config.py:
```diff
@@ -810,45 +810,30 @@ def _s3_secret_access_key():
-        Used as the background for most widgets. Examples of widgets with this
-        background are st.sidebar, st.text_input, st.date_input.
-    """,
+    description="Background color used for the sidebar and most interactive widgets.",
```
most **of the** interactive widgets? (if you decide to change, I've seen this text in a few places)

kantuni replied:
I might be wrong here - feel free to ignore this comment.

vdonato replied:
I think the two are grammatically equivalent, so I'll leave this as-is as the wording came from product. I also think there will probably be one last pass on the copy before release where any small tweaks can be made

vdonato committed:
Remove unnoticed reference to theme.name

This one managed to sneak by since it has no effect on anything in the
case where theme.name is never set.

vdonato committed:
Address PR comments

vdonato committed:
Get rid of an unnecessary import
